tests/integration/test_fim/test_synchronization/test_sync_overlap.py

```diff
@@ -0,0 +1,180 @@
+wazuh_log_monitor = FileMonitor(LOG_FILE_PATH)
+
+
+# Tests
+@pytest.mark.parametrize('configuration, metadata', zip(configurations, configuration_metadata), ids=test_case_ids)
+@pytest.mark.parametrize('files_number', [configuration_metadata[0]['files']])
+def test_sync_overlap(configuration, metadata, set_wazuh_configuration_fim, create_files_in_folder,
+                      restart_syscheck_function, wait_for_fim_start_function):
+    '''
+    description: Check if the 'wazuh-syscheckd' daemon is performing a synchronization at the interval specified in the
+                 configuration, using the 'interval' tag, if a new synchronization is fired, and the last sync message
+                 has been recieved in less time than 'response_timeout, the sync interval is doubled.
+                 The new value for interval cannot be higher than max_interval option. After a new sync interval is
+                 tried and the last message was recieved in a time that is higher than response_timeout, the sync
+                 interval value is returned to the configured value.
+
+    test_phases:
+        - Create a folder with a number of files inside.
+        - Restart syscheckd.
+        - Check that a sync interval started.
+        - Check that next sync is skipped and interval value is doubled
+        - Check that interval value is returned to configured value after successful sync
+
+    wazuh_min_version: 4.5.0
+
+    tier: 2
+
+    parameters:
+        - configuration:
+            type: dict
+            brief: Configuration values for ossec.conf.
+        - metadata:
+            type: dict
+            brief: Test case data.
+        - set_wazuh_configuration_fim:
+            type: fixture
+            brief: Set ossec.conf and local_internal_options configuration.
+        - create_files_in_folder:
+            type: fixture
+            brief: create files in monitored folder, and delete them after the test.
+        - restart_syscheck_function:
+            type: fixture
+            brief: restart syscheckd daemon, and truncate the ossec.log.
+        - wait_for_fim_start_function:
+            type: fixture
+            brief: check that the starting fim scan is detected.
+
+    assertions:
+        - Verify that the new value for interval when doubled is equal or lower to max_interval.
+
+    input_description: A test case (sync_interval) is contained in external YAML file (cases_sync_overlap.yaml) which
+                       includes configuration settings for the 'wazuh-syscheckd' daemon. That is combined with
+                       the interval periods and the testing directory to be monitored defined in this module.
+
+    expected_output:
+        - r'Initializing FIM Integrity Synchronization check'
+        - r"*Sync still in progress. Skipped next sync and increased interval.*'(\\d+)s'"
+        - r".*Previous sync was successful. Sync interval is reset to: '(\\d+)s'"
+
+    tags:
+        - scheduled
```

Comment:
What happens with real-time and who-data mode?

Reply:
This test is only for scheduled. 
It checks the syncs which are only done when the changes are detected on scheduled. realtime and whodata require no sync since it is automatically sent to the db.